File: 1342 Number of Steps to Reduce a Number to Zero.py
class Solution:
    def numberOfSteps (self, num: int) -> int:
        return bin(num).count('1') + len(bin(num)) - 3
'''
# another solution
class Solution:
    def numberOfSteps (self, num: int) -> int:
        res = 0
        while num != 0:
            res += (num & 1) + 1
            num >>= 1
        return res - 1

# from leetcode
class Solution:
    def numberOfSteps (self, num: int) -> int:
        step = 0
        while num != 0:
            if num % 2 == 0:
                num /= 2
            else:
                num -= 1
            step += 1
            
        return step
'''

File: 1342 Number of Steps to Reduce a Number to Zero.java
class Solution {
    public int numberOfSteps (int num) {
        return Integer.bitCount(num) + Integer.toBinaryString(num).length() - 1;
    }
}

/* from leetcode
class Solution {
    public int numberOfSteps (int num) {
        int steps = 0;        
        while (num > 0) {
            if (num % 2 == 0) {
                num /= 2;
            } else {
                num -= 1;
            }            
            steps++;
        }        
        return steps;
    }
}
*/